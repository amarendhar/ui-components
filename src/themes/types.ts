export enum PALETTE_MODE {
  LIGHT = 'light',
  DARK = 'dark'
}

export interface Color {
  50: string
  100: string
  200: string
  300: string
  400: string
  500: string
  600: string
  700: string
  800: string
  900: string
  A100: string
  A200: string
  A400: string
  A700: string
}

// use standalone interface over typeof colors/commons
// to enable module augmentation
export interface CommonColors {
  black: string
  white: string
}

export type ColorPartial = Partial<Color>

export type PaletteColorOptions = SimplePaletteColorOptions | ColorPartial

// Note: Material-UI type of 'SimplePaletteColorOptions'
// export interface SimplePaletteColorOptions {
//   light?: string
//   main: string
//   dark?: string
//   contrastText?: string
// }

// Note: ui-library type of 'SimplePaletteColorOptions'
export type SimplePaletteColorOptions = string

export interface PaletteColor {
  light: string
  main: string
  dark: string
  contrastText: string
}

export type PaletteTonalOffset =
  | number
  | {
      light: number
      dark: number
    }

export interface TypeText {
  primary: string
  secondary: string
  disabled: string
}

export interface TypeAction {
  active: string
  hover: string
  hoverOpacity: number
  selected: string
  selectedOpacity: number
  disabled: string
  disabledOpacity: number
  disabledBackground: string
  focus: string
  focusOpacity: number
  activatedOpacity: number
}

export interface TypeBackground {
  default: string
  paper: string
}

export type TypeDivider = string

export interface PaletteOptions {
  common?: Partial<CommonColors>
  mode?: PALETTE_MODE
  primary?: PaletteColorOptions
  secondary?: PaletteColorOptions
  error?: PaletteColorOptions
  warning?: PaletteColorOptions
  info?: PaletteColorOptions
  success?: PaletteColorOptions
  grey?: ColorPartial
  text?: Partial<TypeText>
  // contrastThreshold?: number
  // tonalOffset?: PaletteTonalOffset
  // divider?: string
  // action?: Partial<TypeAction>
  // background?: Partial<TypeBackground>
  // getContrastText?: (background: string) => string
  // augmentColor: (options: PaletteAugmentColorOptions) => PaletteColor;
}

export interface Palette {
  common: CommonColors
  mode: PALETTE_MODE
  primary: PaletteColor
  secondary: PaletteColor
  error: PaletteColor
  warning: PaletteColor
  info: PaletteColor
  success: PaletteColor
  grey: Color
  text: TypeText
  // contrastThreshold: number;
  // tonalOffset: PaletteTonalOffset;
  // divider: TypeDivider;
  // action: TypeAction;
  // background: TypeBackground;
  // getContrastText: (background: string) => string;
  // augmentColor: (options: PaletteAugmentColorOptions) => PaletteColor;
}
