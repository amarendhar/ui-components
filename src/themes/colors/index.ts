const common = {
  black: '#000',
  white: '#fff',
}

const grey = {
  50: '#fafafa',
  100: '#f5f5f5',
  200: '#eeeeee',
  300: '#e0e0e0',
  400: '#bdbdbd',
  500: '#9e9e9e',
  600: '#757575',
  700: '#616161',
  800: '#424242',
  900: '#212121',
  A100: '#f5f5f5',
  A200: '#eeeeee',
  A400: '#bdbdbd',
  A700: '#616161',
}

const purple = {
  50: '#f3e5f5',
  100: '#e1bee7',
  200: '#ce93d8',
  300: '#ba68c8',
  400: '#ab47bc',
  500: '#9c27b0',
  600: '#8e24aa',
  700: '#7b1fa2',
  800: '#6a1b9a',
  900: '#4a148c',
  A100: '#ea80fc',
  A200: '#e040fb',
  A400: '#d500f9',
  A700: '#aa00ff',
}

const red = {
  50: '#ffebee',
  100: '#ffcdd2',
  200: '#ef9a9a',
  300: '#e57373',
  400: '#ef5350',
  500: '#f44336',
  600: '#e53935',
  700: '#d32f2f',
  800: '#c62828',
  900: '#b71c1c',
  A100: '#ff8a80',
  A200: '#ff5252',
  A400: '#ff1744',
  A700: '#d50000',
}

const orange = {
  50: '#fff3e0',
  100: '#ffe0b2',
  200: '#ffcc80',
  300: '#ffb74d',
  400: '#ffa726',
  500: '#ff9800',
  600: '#fb8c00',
  700: '#f57c00',
  800: '#ef6c00',
  900: '#e65100',
  A100: '#ffd180',
  A200: '#ffab40',
  A400: '#ff9100',
  A700: '#ff6d00',
}

const blue = {
  50: '#e3f2fd',
  100: '#bbdefb',
  200: '#90caf9',
  300: '#64b5f6',
  400: '#42a5f5',
  500: '#2196f3',
  600: '#1e88e5',
  700: '#1976d2',
  800: '#1565c0',
  900: '#0d47a1',
  A100: '#82b1ff',
  A200: '#448aff',
  A400: '#2979ff',
  A700: '#2962ff',
}

const lightBlue = {
  50: '#e1f5fe',
  100: '#b3e5fc',
  200: '#81d4fa',
  300: '#4fc3f7',
  400: '#29b6f6',
  500: '#03a9f4',
  600: '#039be5',
  700: '#0288d1',
  800: '#0277bd',
  900: '#01579b',
  A100: '#80d8ff',
  A200: '#40c4ff',
  A400: '#00b0ff',
  A700: '#0091ea',
}

const green = {
  50: '#e8f5e9',
  100: '#c8e6c9',
  200: '#a5d6a7',
  300: '#81c784',
  400: '#66bb6a',
  500: '#4caf50',
  600: '#43a047',
  700: '#388e3c',
  800: '#2e7d32',
  900: '#1b5e20',
  A100: '#b9f6ca',
  A200: '#69f0ae',
  A400: '#00e676',
  A700: '#00c853',
}

export type ColorType = {
  50: string
  100: string
  200: string
  300: string
  400: string
  500: string
  600: string
  700: string
  800: string
  900: string
  A100: string
  A200: string
  A400: string
  A700: string
}

export type Colors = {
  common: {
    black: string
    white: string
  }
  grey: ColorType
  purple: ColorType
  red: ColorType
  orange: ColorType
  blue: ColorType
  lightBlue: ColorType
  green: ColorType
}

const colors: Colors = {
  common,
  grey,
  purple,
  red,
  orange,
  blue,
  lightBlue,
  green,
}

export default colors
