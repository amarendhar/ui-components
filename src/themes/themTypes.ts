export type Theme = {
  space: Record<'sm' | 'md' | 'lg', number>
  fontSizes: Record<'sm' | 'md' | 'lg', number>
  radii: Record<'sm' | 'md' | 'lg', number>
  palette: Palette
  breakpoints: Record<'xs' | 'sm' | 'md' | 'lg' | 'xl', number>
  mediaQuery: Record<
    | 'xs'
    | 'sm'
    | 'md'
    | 'lg'
    | 'xl'
    | '<xs'
    | '<sm'
    | '<md'
    | '<lg'
    | '<xl'
    | 'xsQuery'
    | 'smQuery'
    | 'mdQuery'
    | 'lgQuery'
    | 'xlQuery'
    | '<xsQuery'
    | '<smQuery'
    | '<mdQuery'
    | '<lgQuery'
    | '<xlQuery',
    string
  >
}

export type DisabledPaletteColor = {
  main: string
  contrastText: string
}

/**
 * Variants referred from react-bootstrap
 *  https://react-bootstrap.github.io/components/buttons/
 */

export enum CommonColors {
  // ToDo: is default-type required ?
  default = 'default',
  primary = 'primary',
  secondary = 'secondary',
  error = 'error',
  info = 'info',
  success = 'success',
  warning = 'warning',
  // ToDo: is disable-type required ?
  // disabled = 'disabled',
}

export enum CommonSizes {
  sm = 'sm',
  md = 'md',
  lg = 'lg',
}

export enum CommonVariants {
  contained = 'contained',
  outlined = 'outlined',
  text = 'text',
}

/**
 * Types Referred from material-ui
 */
export enum PALETTE_MODE {
  LIGHT = 'light',
  DARK = 'dark',
}

export interface Color {
  50: string
  100: string
  200: string
  300: string
  400: string
  500: string
  600: string
  700: string
  800: string
  900: string
  A100: string
  A200: string
  A400: string
  A700: string
}

// use standalone interface over typeof colors/commons
// to enable module augmentation
export interface CommonPaletteColors {
  black: string
  white: string
}

export type ColorPartial = Partial<Color>

// Note: Material-UI type of 'PaletteColorOptions'
// export type PaletteColorOptions = SimplePaletteColorOptions | ColorPartial

// Note: Material-UI type of 'SimplePaletteColorOptions'
// export interface SimplePaletteColorOptions {
//   light?: string
//   main: string
//   dark?: string
//   contrastText?: string
// }

export type PaletteColorOptions = string

export interface PaletteColor {
  main: string
  light: string
  dark: string
  contrastText: string
}

export type PaletteTonalOffset =
  | number
  | {
      light: number
      dark: number
    }

export interface TypeText {
  primary: string
  secondary: string
  disabled: string
  hint?: string
}

export interface TypeAction {
  active: string
  hover: string
  hoverOpacity: number
  selected: string
  selectedOpacity: number
  disabled: string
  disabledOpacity: number
  disabledBackground: string
  focus: string
  focusOpacity: number
  activatedOpacity: number
}

export interface TypeBackground {
  default: string
  paper: string
}

export type TypeDivider = string

export interface PaletteOptions {
  common?: Partial<CommonPaletteColors>
  mode?: PALETTE_MODE
  default?: PaletteColorOptions
  primary?: PaletteColorOptions
  secondary?: PaletteColorOptions
  error?: PaletteColorOptions
  info?: PaletteColorOptions
  success?: PaletteColorOptions
  warning?: PaletteColorOptions
  grey?: ColorPartial
  text?: Partial<TypeText>
  // contrastThreshold?: number
  // tonalOffset?: PaletteTonalOffset
  // divider?: string
  // action?: Partial<TypeAction>
  // background?: Partial<TypeBackground>
  // getContrastText?: (background: string) => string
  // augmentColor: (options: PaletteAugmentColorOptions) => PaletteColor;
}

export type CommonPalette = {
  default: PaletteColor
  primary: PaletteColor
  secondary: PaletteColor
  error: PaletteColor
  info: PaletteColor
  success: PaletteColor
  warning: PaletteColor
}

export interface Palette extends CommonPalette {
  common: CommonPaletteColors
  mode: PALETTE_MODE
  disabled: DisabledPaletteColor
  // ToDo: CommonVariants.contained is required ?
  [CommonVariants.contained]: CommonPalette
  [CommonVariants.outlined]: CommonPalette
  [CommonVariants.text]: CommonPalette
  textt?: TypeText
  grey: Color
  // contrastThreshold: number;
  // tonalOffset: PaletteTonalOffset;
  // divider: TypeDivider;
  // action: TypeAction;
  // background: TypeBackground;
  // getContrastText: (background: string) => string;
  // augmentColor: (options: PaletteAugmentColorOptions) => PaletteColor;
}

export type DecomposedColor = {
  type: string
  rgbValues: number[]
}
