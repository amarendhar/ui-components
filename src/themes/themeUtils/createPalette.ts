import { PaletteOptions } from 'themes/types'
import colors from '../colors'
import {
  decomposeColor,
  clamp,
  getContrastRatio,
  lighten,
  darken,
} from './colorManipulator'

const tonalOffset = 0.2
const contrastThreshold = 3

/**
 * ToDo: Do not use type 'any'
 */
/**
 * Refer https://bareynol.github.io/mui-theme-creator
 */
/**
 * Refer `light` method from @material-ui/core/styles/createPalette.js
 */
export const light = {
  // The colors used to style the text.
  text: {
    // The most important text.
    primary: 'rgba(0, 0, 0, 0.87)',
    // Secondary text.
    secondary: 'rgba(0, 0, 0, 0.6)',
    // Disabled text have even lower visual prominence.
    disabled: 'rgba(0, 0, 0, 0.38)',
  },
  // The color used to divide different elements.
  divider: 'rgba(0, 0, 0, 0.12)',
  // The background colors used to style the surfaces.
  // Consistency between these values is important.
  background: {
    paper: colors.common.white,
    default: colors.common.white,
  },
  // The colors used to style the action elements.
  action: {
    // The color of an active action like an icon button.
    active: 'rgba(0, 0, 0, 0.54)',
    // The color of an hovered action.
    hover: 'rgba(0, 0, 0, 0.04)',
    hoverOpacity: 0.04,
    // The color of a selected action.
    selected: 'rgba(0, 0, 0, 0.08)',
    selectedOpacity: 0.08,
    // The color of a disabled action.
    disabled: 'rgba(0, 0, 0, 0.26)',
    // The background color of a disabled action.
    disabledBackground: 'rgba(0, 0, 0, 0.12)',
    disabledOpacity: 0.38,
    focus: 'rgba(0, 0, 0, 0.12)',
    focusOpacity: 0.12,
    activatedOpacity: 0.12,
  },
}

/**
 * Refer `dark` method from @material-ui/core/styles/createPalette.js
 */
export const dark = {
  text: {
    primary: colors.common.white,
    secondary: 'rgba(255, 255, 255, 0.7)',
    disabled: 'rgba(255, 255, 255, 0.5)',
    icon: 'rgba(255, 255, 255, 0.5)',
  },
  divider: 'rgba(255, 255, 255, 0.12)',
  background: {
    paper: '#121212',
    default: '#121212',
  },
  action: {
    active: colors.common.white,
    hover: 'rgba(255, 255, 255, 0.08)',
    hoverOpacity: 0.08,
    selected: 'rgba(255, 255, 255, 0.16)',
    selectedOpacity: 0.16,
    disabled: 'rgba(255, 255, 255, 0.3)',
    disabledBackground: 'rgba(255, 255, 255, 0.12)',
    disabledOpacity: 0.38,
    focus: 'rgba(255, 255, 255, 0.12)',
    focusOpacity: 0.12,
    activatedOpacity: 0.24,
  },
}

/**
 * Refer `getContrastText` method from @material-ui/core/styles/createPalette.js
 */
const getContrastText = (background: any) => {
  const contrastText =
    getContrastRatio(background, dark.text.primary) >= contrastThreshold
      ? dark.text.primary
      : light.text.primary

  if (process.env.NODE_ENV !== 'production') {
    const contrast = getContrastRatio(background, contrastText)

    if (contrast < 3) {
      console.error(
        [
          `UI-Library: The contrast ratio of ${contrast}:1 for ${contrastText} on ${background}`,
          'falls below the WCAG recommended absolute minimum contrast ratio of 3:1.',
          'https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-contrast',
        ].join('\n')
      )
    }
  }

  return contrastText
}

/**
 * Refer `augmentColor` method from @material-ui/core/styles/createPalette.js
 */
const getColorStates = (color: any) => {
  // Note: Do not use common variable like below, it's not giving correct 'dark' value, when used common variable like that.
  // const decomposedColor = decomposeColor(color)
  
  return {
    main: color,
    light: lighten({
      color: decomposeColor(color),
      coefficient: clamp(tonalOffset),
    }),
    dark: darken({
      color: decomposeColor(color),
      coefficient: clamp(tonalOffset * 1.5),
    }),
    contrastText: getContrastText(color),
  }
}


/**
 * Refer `paletteOutput` object from @material-ui/core/styles/createPalette.js
 */
const createPalette = (palette: PaletteOptions = {}) => {
  const mode = palette?.mode || 'light'
  const {
    primary = mode === 'dark' ? colors.blue[200] : colors.blue[700],
    secondary = mode === 'dark' ? colors.purple[200] : colors.purple[500],
    error = mode === 'dark' ? colors.red[500] : colors.red[700],
    info = mode === 'dark' ? colors.lightBlue[400] : colors.lightBlue[700],
    success = mode === 'dark' ? colors.green[400] : colors.green[800],
    warning = mode === 'dark' ? colors.orange[400] : '#ED6C02',
  } = palette

  const paletteOutput = {
    // A collection of common colors.
    common: colors.common,
    // The palette mode, can be light or dark.
    mode,
    // The colors used to represent primary interface elements for a user.
    primary: getColorStates(primary),
    secondary: getColorStates(secondary),
    error: getColorStates(error),
    info: getColorStates(info),
    success: getColorStates(success),
    warning: getColorStates(warning),
  }

  return paletteOutput
}

export default createPalette
