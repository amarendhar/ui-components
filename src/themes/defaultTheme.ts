import { Theme } from './themTypes'

export const space: Theme['space'] = {
  sm: 5,
  md: 10,
  lg: 15,
}

export const fontSizes: Theme['fontSizes'] = {
  sm: 13,
  md: 14,
  lg: 15,
}

export const fontWeights: Record<'regular' | 'bold', number> = {
  regular: 400,
  bold: 600,
}

export const radii: Theme['radii'] = {
  sm: 3,
  md: 6,
  lg: 10,
}

const shadows: Record<'header' | 'box', string> = {
  header: '-5px 5px 5px -5px rgb(0 0 0)',
  box: '0px 3px 5px 0px rgb(0 0 0 / 20%)',
}

// ToDo: Remove this, this is old code
export const breakpoints: Theme['breakpoints'] = {
  xs: 600, // <= 600, Extra small devices (phones, 600px and down)
  sm: 600, // >= 600, Small devices (portrait tablets and large phones, 600px and up)
  md: 768, // >= 768, Medium devices (landscape tablets, 768px and up)
  lg: 992, // >= 1024, Large devices (laptops/desktops, 992px and up)
  xl: 1200, // >= 1200, Extra large devices (large laptops and desktops, 1200px and up)
}

export const mediaQuery: Theme['mediaQuery'] = {
  xs: `@media (min-width: ${breakpoints.xs}px)`,
  sm: `@media (min-width: ${breakpoints.sm}px)`,
  md: `@media (min-width: ${breakpoints.md}px)`,
  lg: `@media (min-width: ${breakpoints.lg}px)`,
  xl: `@media (min-width: ${breakpoints.xl}px)`,
  '<xs': `@media (max-width: ${breakpoints.xs}px)`,
  '<sm': `@media (max-width: ${breakpoints.sm}px)`,
  '<md': `@media (max-width: ${breakpoints.md}px)`,
  '<lg': `@media (max-width: ${breakpoints.lg}px)`,
  '<xl': `@media (max-width: ${breakpoints.xl}px)`,
  get xsQuery() {
    return this.xs.replace('@media', '')
  },
  get smQuery() {
    return this.sm.replace('@media', '')
  },
  get mdQuery() {
    return this.md.replace('@media', '')
  },
  get lgQuery() {
    return this.lg.replace('@media', '')
  },
  get xlQuery() {
    return this.xl.replace('@media', '')
  },
  get '<xsQuery'() {
    return this['<xs'].replace('@media', '')
  },
  get '<smQuery'() {
    return this['<sm'].replace('@media', '')
  },
  get '<mdQuery'() {
    return this['<md'].replace('@media', '')
  },
  get '<lgQuery'() {
    return this['<lg'].replace('@media', '')
  },
  get '<xlQuery'() {
    return this['<xl'].replace('@media', '')
  },
}

const defaultTheme: Omit<Theme, 'palette'> = {
  space,
  fontSizes,
  fontWeights,
  radii,
  shadows,
  /**
   * Do not include custom-breakpoints like below, when using styled-system library
   *  to avoid conflicts against breakpoints from styled-system npm,
   *  otherwise styled-system will throw error, and will not render UI in storybook.
   */
  // breakpoints,
  breakpoints,
  mediaQuery,
}

export default defaultTheme
