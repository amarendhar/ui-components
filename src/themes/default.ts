import { ObjType } from 'types'
import { createPalette } from './themeUtils'

export const colors = {
  white: {
    100: '#ffffff',
    80: '#ecedef',
  },
  blue: {
    100: '#115293',
    80: '#1976d2',
  },
  green: {
    100: '#367f39',
    80: '#4caf50',
  },
  orange: {
    100: '#cb7d09',
    80: '#ff9800',
  },
  red: {
    100: '#9a0036',
    80: '#dc004e',
  },
  grey: {
    100: '#808080',
    80: '#969494',
  },
  silver: {
    100: '#c5c2c2',
    80: '#e0e0e0',
    60: '#efefef',
  },
  black: {
    100: '#000000',
    80: '#484848',
  },
  yellow: {
    100: '#F8E831',
  },
  outline: {
    100: '#e0e0e0',
    80: 'transparent',
  },
}

export const space = {
  small: 5,
  medium: 10,
  large: 15,
}

export const fontSize = {
  small: 13,
  medium: 14,
  large: 15,
}

export const paddingFactor = {
  small: 4,
  medium: 6,
  large: 8,
}

export const breakpoints = {
  sm: 767, // Phones (767/75 i.e. 7.7 to 10.0 inch) -> "<=767 OR < 768"
  md: 768, // Tablets (>= 10 inch) -> ">= 768"
  lg: 992, // Laptops (>= 13 inch) -> ">= 992"
}

export const mediaQuery: ObjType = {
  sm: `@media (max-width: ${breakpoints.sm}px)`, // Phones "<=767 OR < 768"
  md: `@media (min-width: ${breakpoints.md}px)`, // Tablets ">= 768"
  lg: `@media (min-width: ${breakpoints.lg}px)`, // Laptops ">= 992"
  // ">=" i.e. for min-width's
  '>sm': `@media (min-width: 576px)`, // ">= 576"
  '>md': `@media (min-width: 768px)`, // ">= 768"
  '>lg': `@media (min-width: 992px)`, // ">= 992"
  // "<=" i.e. for max-width's
  '<sm': `@media (max-width: 767px)`, // "< 768"
  '<md': `@media (max-width: 991px)`, // "< 991"
  '<lg': `@media (max-width: 1199px)`, // "< 1200"
}

const palette = createPalette()

export const defaultTheme = {
  /**
   * Do not include custom-breakpoints like below,
   *  to avoid conflicts against breakpoints from styled-system npm,
   *  otherwise styled-system will throw error, and will not render UI in storybook.
   */
  // breakpoints,
  mediaQuery,
  space,
  fontSize,
  paddingFactor,
  colors,
  palette,
  default: {
    bg: colors.silver,
    border: colors.silver,
    text: colors.black,
    outline: {
      bg: colors.outline,
      border: colors.black,
      text: colors.black,
    },
  },
  primary: {
    bg: colors.blue,
    border: colors.blue,
    text: colors.white,
    outline: {
      bg: colors.outline,
      border: colors.blue,
      text: colors.blue,
    },
  },
  success: {
    bg: colors.green,
    border: colors.green,
    text: colors.white,
    outline: {
      bg: colors.outline,
      border: colors.green,
      text: colors.green,
    },
  },
  warning: {
    bg: colors.orange,
    border: colors.orange,
    text: colors.white,
    outline: {
      bg: colors.outline,
      border: colors.orange,
      text: colors.orange,
    },
  },
  danger: {
    bg: colors.red,
    border: colors.red,
    text: colors.white,
    outline: {
      bg: colors.outline,
      border: colors.red,
      text: colors.red,
    },
  },
  disabled: {
    bg: colors.silver,
    border: colors.silver,
    text: colors.black,
    outline: {
      bg: colors.outline,
      border: colors.silver,
      text: colors.black,
    },
  },
}
