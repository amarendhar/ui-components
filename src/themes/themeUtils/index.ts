import createPalette from './createPalette'
import variants from './variants'

export {
  createPalette,
  variants
}
