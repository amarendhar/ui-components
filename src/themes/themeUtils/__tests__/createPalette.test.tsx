import createPalette from '../createPalette'
import { PALETTE_MODE } from 'themes/themTypes'

describe('createPalette', () => {
  it('Should return palette for light-mode', () => {
    const palette = createPalette()

    expect(palette).toEqual({
      ...lightPalette,
      mode: PALETTE_MODE.LIGHT,
    })

    expect(palette).toMatchSnapshot()
  })

  it('Should return palette for dark-mode', () => {
    const palette = createPalette({
      mode: PALETTE_MODE.DARK,
    })

    expect(palette).toEqual({
      ...darkPalette,
      mode: PALETTE_MODE.DARK,
    })

    expect(palette).toMatchSnapshot()
  })

  /**
   * For Color codes refer
   *  https://material.io/resources/color/#!/?view.left=0&view.right=0&primary.color=FF9800&secondary.color=2196F3
   *
   * where '#e65100' is orange[500]
   * where '#2196f3' is blue[500]
   */
  it('Should return palette for primary-orange & secondary-blue', () => {
    const palette = createPalette({
      primary: '#e65100',
      secondary: '#2196f3',
    })

    expect(palette).toEqual({
      ...primaryOrangeAndSecondaryBluePalette,
      mode: PALETTE_MODE.LIGHT,
    })

    expect(palette).toMatchSnapshot()
  })

  const commonPalette = {
    common: {
      black: '#000',
      white: '#fff',
    },
  }

  const lightPalette = {
    ...commonPalette,
    error: {
      contrastText: '#fff',
      dark: 'rgb(147, 32, 32)',
      light: 'rgb(219, 88, 88)',
      main: '#d32f2f',
    },
    info: {
      contrastText: '#fff',
      dark: 'rgb(1, 95, 146)',
      light: 'rgb(52, 159, 218)',
      main: '#0288d1',
    },
    primary: {
      contrastText: '#fff',
      dark: 'rgb(17, 82, 147)',
      light: 'rgb(71, 145, 219)',
      main: '#1976d2',
    },
    secondary: {
      contrastText: '#fff',
      dark: 'rgb(109, 27, 123)',
      light: 'rgb(175, 82, 191)',
      main: '#9c27b0',
    },
    success: {
      contrastText: '#fff',
      dark: 'rgb(32, 87, 35)',
      light: 'rgb(87, 151, 91)',
      main: '#2e7d32',
    },
    warning: {
      contrastText: '#fff',
      dark: 'rgb(165, 75, 1)',
      light: 'rgb(240, 137, 52)',
      main: '#ED6C02',
    },
  }

  const darkPalette = {
    ...commonPalette,
    error: {
      contrastText: '#fff',
      dark: 'rgb(170, 46, 37)',
      light: 'rgb(246, 104, 94)',
      main: '#f44336',
    },
    info: {
      contrastText: 'rgba(0, 0, 0, 0.87)',
      dark: 'rgb(28, 127, 172)',
      light: 'rgb(83, 196, 247)',
      main: '#29b6f6',
    },
    primary: {
      contrastText: 'rgba(0, 0, 0, 0.87)',
      dark: 'rgb(100, 141, 174)',
      light: 'rgb(166, 212, 250)',
      main: '#90caf9',
    },
    secondary: {
      contrastText: 'rgba(0, 0, 0, 0.87)',
      dark: 'rgb(144, 102, 151)',
      light: 'rgb(215, 168, 223)',
      main: '#ce93d8',
    },
    success: {
      contrastText: 'rgba(0, 0, 0, 0.87)',
      dark: 'rgb(71, 130, 74)',
      light: 'rgb(132, 200, 135)',
      main: '#66bb6a',
    },
    warning: {
      contrastText: 'rgba(0, 0, 0, 0.87)',
      dark: 'rgb(178, 116, 26)',
      light: 'rgb(255, 184, 81)',
      main: '#ffa726',
    },
  }

  const primaryOrangeAndSecondaryBluePalette = {
    ...commonPalette,
    error: {
      contrastText: '#fff',
      dark: 'rgb(147, 32, 32)',
      light: 'rgb(219, 88, 88)',
      main: '#d32f2f',
    },
    info: {
      contrastText: '#fff',
      dark: 'rgb(1, 95, 146)',
      light: 'rgb(52, 159, 218)',
      main: '#0288d1',
    },
    primary: {
      contrastText: '#fff',
      dark: 'rgb(161, 56, 0)',
      light: 'rgb(235, 115, 51)',
      main: '#e65100',
    },
    secondary: {
      contrastText: '#fff',
      dark: 'rgb(23, 105, 170)',
      light: 'rgb(77, 171, 245)',
      main: '#2196f3',
    },
    success: {
      contrastText: '#fff',
      dark: 'rgb(32, 87, 35)',
      light: 'rgb(87, 151, 91)',
      main: '#2e7d32',
    },
    warning: {
      contrastText: '#fff',
      dark: 'rgb(165, 75, 1)',
      light: 'rgb(240, 137, 52)',
      main: '#ED6C02',
    },
  }
})
