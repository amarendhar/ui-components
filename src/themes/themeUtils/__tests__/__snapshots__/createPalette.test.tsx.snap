// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPalette Should return palette for dark-mode 1`] = `
Object {
  "common": Object {
    "black": "#000",
    "white": "#fff",
  },
  "error": Object {
    "contrastText": "#fff",
    "dark": "rgb(170, 46, 37)",
    "light": "rgb(246, 104, 94)",
    "main": "#f44336",
  },
  "info": Object {
    "contrastText": "rgba(0, 0, 0, 0.87)",
    "dark": "rgb(28, 127, 172)",
    "light": "rgb(83, 196, 247)",
    "main": "#29b6f6",
  },
  "mode": "dark",
  "primary": Object {
    "contrastText": "rgba(0, 0, 0, 0.87)",
    "dark": "rgb(100, 141, 174)",
    "light": "rgb(166, 212, 250)",
    "main": "#90caf9",
  },
  "secondary": Object {
    "contrastText": "rgba(0, 0, 0, 0.87)",
    "dark": "rgb(144, 102, 151)",
    "light": "rgb(215, 168, 223)",
    "main": "#ce93d8",
  },
  "success": Object {
    "contrastText": "rgba(0, 0, 0, 0.87)",
    "dark": "rgb(71, 130, 74)",
    "light": "rgb(132, 200, 135)",
    "main": "#66bb6a",
  },
  "warning": Object {
    "contrastText": "rgba(0, 0, 0, 0.87)",
    "dark": "rgb(178, 116, 26)",
    "light": "rgb(255, 184, 81)",
    "main": "#ffa726",
  },
}
`;

exports[`createPalette Should return palette for light-mode 1`] = `
Object {
  "common": Object {
    "black": "#000",
    "white": "#fff",
  },
  "error": Object {
    "contrastText": "#fff",
    "dark": "rgb(147, 32, 32)",
    "light": "rgb(219, 88, 88)",
    "main": "#d32f2f",
  },
  "info": Object {
    "contrastText": "#fff",
    "dark": "rgb(1, 95, 146)",
    "light": "rgb(52, 159, 218)",
    "main": "#0288d1",
  },
  "mode": "light",
  "primary": Object {
    "contrastText": "#fff",
    "dark": "rgb(17, 82, 147)",
    "light": "rgb(71, 145, 219)",
    "main": "#1976d2",
  },
  "secondary": Object {
    "contrastText": "#fff",
    "dark": "rgb(109, 27, 123)",
    "light": "rgb(175, 82, 191)",
    "main": "#9c27b0",
  },
  "success": Object {
    "contrastText": "#fff",
    "dark": "rgb(32, 87, 35)",
    "light": "rgb(87, 151, 91)",
    "main": "#2e7d32",
  },
  "warning": Object {
    "contrastText": "#fff",
    "dark": "rgb(165, 75, 1)",
    "light": "rgb(240, 137, 52)",
    "main": "#ED6C02",
  },
}
`;

exports[`createPalette Should return palette for primary-orange & secondary-blue 1`] = `
Object {
  "common": Object {
    "black": "#000",
    "white": "#fff",
  },
  "error": Object {
    "contrastText": "#fff",
    "dark": "rgb(147, 32, 32)",
    "light": "rgb(219, 88, 88)",
    "main": "#d32f2f",
  },
  "info": Object {
    "contrastText": "#fff",
    "dark": "rgb(1, 95, 146)",
    "light": "rgb(52, 159, 218)",
    "main": "#0288d1",
  },
  "mode": "light",
  "primary": Object {
    "contrastText": "#fff",
    "dark": "rgb(161, 56, 0)",
    "light": "rgb(235, 115, 51)",
    "main": "#e65100",
  },
  "secondary": Object {
    "contrastText": "#fff",
    "dark": "rgb(23, 105, 170)",
    "light": "rgb(77, 171, 245)",
    "main": "#2196f3",
  },
  "success": Object {
    "contrastText": "#fff",
    "dark": "rgb(32, 87, 35)",
    "light": "rgb(87, 151, 91)",
    "main": "#2e7d32",
  },
  "warning": Object {
    "contrastText": "#fff",
    "dark": "rgb(165, 75, 1)",
    "light": "rgb(240, 137, 52)",
    "main": "#ED6C02",
  },
}
`;
